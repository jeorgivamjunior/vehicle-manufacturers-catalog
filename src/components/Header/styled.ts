import { AppBar, styled, Toolbar, Typography } from '@mui/material';

export const ToolbarContainer = styled(Toolbar)`
  flex-direction: column;
`;

export const HeaderContainer = styled(AppBar)`
  background-color: #e6f2ce;
`;

export const Image = styled('img')`
  width: 170px;
  padding: 20px;
`;

export const Title = styled(Typography)`
  padding-bottom: 30px;
  font-weight: bold;
  color: #7cb800;
`;
