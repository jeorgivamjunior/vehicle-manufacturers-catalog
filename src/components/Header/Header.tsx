import { FC } from 'react';

import factoryImage from '../../assets/car.png';
import { HeaderContainer, ToolbarContainer, Title, Image } from './styled';

export const Header: FC = () => {
  return (
    <HeaderContainer position="relative">
      <ToolbarContainer>
        <Image src={factoryImage} />
        <Title variant="h4">Vehicle Manufacturers Catalog</Title>
      </ToolbarContainer>
    </HeaderContainer>
  );
};
