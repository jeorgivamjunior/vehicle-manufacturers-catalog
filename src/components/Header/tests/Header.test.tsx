import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Header } from '../Header';

describe('Header component', () => {
  it('renders the ui correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display the title', () => {
    render(<Header />);
    const title = screen.getByRole('heading');

    expect(title).toBeInTheDocument();
  });

  it('should display the logo', () => {
    render(<Header />);
    const logo = screen.getByRole('img');

    expect(logo).toBeInTheDocument();
  });
});
