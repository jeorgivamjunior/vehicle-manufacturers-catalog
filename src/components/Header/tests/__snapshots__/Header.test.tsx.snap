// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the ui correctly 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-1d22kfw-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-12oimmw-MuiToolbar-root"
  >
    <img
      className="css-z6u0ty"
      src="car.png"
    />
    <h4
      className="MuiTypography-root MuiTypography-h4 css-aziax4-MuiTypography-root"
    >
      Vehicle Manufacturers Catalog
    </h4>
  </div>
</header>
`;
